extension/src/popup/components/account/AccountHeader/index.tsx

```diff
@@ -29,18 +39,24 @@ export const AccountHeader = ({
   publicKey,
 }: AccountHeaderProps) => {
   const { t } = useTranslation();
-  const { isTestnet } = useSelector(settingsNetworkDetailsSelector);
+  const dispatch = useDispatch();
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const networksList = useSelector(settingsNetworksListSelector);
   const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [isNetworkSelectorOpen, setIsNetworkSelectorOpen] = useState(false);
 
-  const dropdownRef = useRef<HTMLUListElement>(null);
+  const calculateModalHeight = (listLength: number) => (listLength + 2) * 6;
+
+  const accountsModalHeight = useRef(0);
+  const networksModalHeight = useRef(0);
```

Comment:
the networks modal is very similar to the accounts modal, so we're going to be reusing a lot of the same logic